src/System.Linq/tests/ConcatTests.cs

```diff
@@ -421,5 +421,89 @@ public void GetEnumerableOfConcatCollectionChainFollowedByEnumerableNodeShouldBe
                 Assert.Equal(0xf00, en.Current);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(GetToArrayDataSources))]
+        public void CollectionInterleavedWithLazyEnumerables_ToArray(IEnumerable<int>[] arrays)
+        {
+            // See https://github.com/dotnet/corefx/issues/23680
+
+            var concats = arrays[0];
```

Comment:
Nit: we try to use var only when the type is obvious from the right hand side, which it isn't here or below